tests/eigen_svd/svd_cdouble_avx512.cpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file
+ *
+ * @copyright
+ * Copyright 2022 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * @test @b eigen_svd_cdouble_avx512
+ * @parblock
+ * This piece of code aims to stress test the FMA execution units of
+ * the CPU, among others, by repetitively solving the singular value
+ * decomposition problem on given input matrices, which involve a lot
+ * of matrix multiplication operations underneath.
+ *
+ * The test is intended to use AVX512 instructions, which are available
+ * with Eigen 3.4 (and up) with AVX512 enabled. The test should not be
+ * created for other configurations (Eigen 3.3.x or GCC pre-12 or below
+ * SKX target)
+ *
+ * The logic comes from the 3rd party library Eigen. The first thread
+ * that gets to run computes a "golden value" of the results, those
+ * being output matrices "U" and "V". Subsequent runs have results
+ * contrasted to those golden values and, whenever they differ, an
+ * error is flagged.
+ *
+ * This particular version of the Eigen SVD tests go for double
+ * precision, complex numbers input matrices and the (divide and
+ * conquer) bi-diagonalization SVD algorithm.
+ *
+ * @note This test requires at least 2 threads to run.
+ * @endparblock
+ */
+
+// Eigen 3.3.8 requires AVX512 vectorization to be enabled.
+// Eigent 3.4.0 has AVX512 enabled by default (if appropriate architecture is requested)
+#define EIGEN_ENABLE_AVX512
```

Comment:
I don't think that this will work with Eigen 3.3.x as it will expose a bug not fixed until 3.4

https://gitlab.com/libeigen/eigen/-/commit/354baa0fb1bb6c2d137288a61339e8cf0e702ad7

This needs to be EIGEN_VECTORIZE_AVX512

which will give us AVX512F but not AVX512DQ (which doesn't work < 3.4).